pkg/cstor/newpool/v1alpha3/build.go

```diff
@@ -0,0 +1,266 @@
+	if len(b.errs) > 0 {
+		return nil, errors.Errorf("%+v", b.errs)
+	}
+	return b.CSP, nil
+}
+
+// WithName sets the Name field of CSP with provided value.
+func (b *Builder) WithName(name string) *Builder {
+	if len(name) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing CSP name"),
+		)
+		return b
+	}
+	b.CSP.Object.Name = name
+	return b
+}
+
+// WithNamespace sets the Namespace field of CSP provided arguments
+func (b *Builder) WithNamespace(namespace string) *Builder {
+	if len(namespace) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing CSP namespace"),
+		)
+		return b
+	}
+	b.CSP.Object.Namespace = namespace
+	return b
+}
+
+// WithAnnotationsNew sets the Annotations field of CSP with provided arguments
+func (b *Builder) WithAnnotationsNew(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing annotations"),
+		)
+		return b
+	}
+	b.CSP.Object.Annotations = make(map[string]string)
+	for key, value := range annotations {
+		b.CSP.Object.Annotations[key] = value
+	}
+	return b
+}
+
+// WithAnnotations appends or overwrites existing Annotations
+// values of CSP with provided arguments
+func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing annotations"),
+		)
+		return b
+	}
+	if b.CSP.Object.Annotations == nil {
+		return b.WithAnnotationsNew(annotations)
+	}
+	for key, value := range annotations {
+		b.CSP.Object.Annotations[key] = value
+	}
+	return b
+}
+
+// WithLabelsNew sets the Labels field of CSP with provided arguments
+func (b *Builder) WithLabelsNew(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing labels"),
+		)
+		return b
+	}
+	b.CSP.Object.Labels = make(map[string]string)
+	for key, value := range labels {
+		b.CSP.Object.Labels[key] = value
+	}
+	return b
+}
+
+// WithLabels appends or overwrites existing Labels
+// values of CSP with provided arguments
+func (b *Builder) WithLabels(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing labels"),
+		)
+		return b
+	}
+	if b.CSP.Object.Labels == nil {
+		return b.WithLabelsNew(labels)
+	}
+	for key, value := range labels {
+		b.CSP.Object.Labels[key] = value
+	}
+	return b
+}
+
+// WithNodeSelector sets the node selector field of CSP with provided argument.
+func (b *Builder) WithNodeSelector(nodeSelector map[string]string) *Builder {
+	if len(nodeSelector) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing nodeSelector"),
+		)
+		return b
+	}
+	b.CSP.Object.Spec.HostName = nodeSelector["kubernetes.io/hostname"]
```

Comment:
there is hardcoding of key.. instead, new copy of nodeSelector can be made and assigned to new member variable NodeSelector..
this spec.HostName has been assigned in WithNodeName (below function) as well

Reply:
yes... missed this. will fix it.